addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightDiscover.java

```diff
@@ -176,44 +183,102 @@ public void sendDiscover(ScheduledExecutorService scheduler) {
         }
 
         resendCounter = 0;
+        this.scheduler = scheduler;
         resendTimer = scheduler.scheduleWithFixedDelay(new SendDiscoverRunnable(), 0, resendTimeoutInMillis,
                 TimeUnit.MILLISECONDS);
     }
 
     @Override
     public void run() {
         try {
-            // logger.debug("Discovery receive thread ready");
-
-            // Now loop forever, waiting to receive packets and printing them.
             while (!willbeclosed) {
                 packet.setLength(buffer.length);
                 datagramSocket.receive(packet);
-                // example: 10.1.1.27,ACCF23F57AD4,HF-LPB100
+                // We expect packets with a format like this: 10.1.1.27,ACCF23F57AD4,HF-LPB100
                 String[] msg = new String(buffer, 0, packet.getLength()).split(",");
-                if (msg.length >= 2 && msg[1].length() == 12) {
-                    // Stop resend timer if we got a packet.
-                    if (resendTimer != null) {
-                        resendTimer.cancel(true);
-                        resendTimer = null;
+
+                if (msg.length != 2 && msg.length != 3) {
+                    // That data packet does not belong to a Milight bridge. Just ignore it.
+                    continue;
+                }
+
+                int version = 3; // Assume version 3
+                // First argument is the IP
+                try {
+                    InetAddress.getByName(msg[0]);
+                } catch (UnknownHostException ignored) {
+                    // That data packet does not belong to a Milight bridge, we expect an IP address as first argument.
+                    // Just ignore it.
+                    continue;
+                }
+
+                // Second argument is the MAC address
+                if (msg[1].length() != 12) {
+                    // That data packet does not belong to a Milight bridge, we expect a MAC address as second argument.
+                    // Just ignore it.
+                    continue;
+                }
+
+                InetAddress addressOfBridge = ((InetSocketAddress) packet.getSocketAddress()).getAddress();
+                if (msg.length == 3) {
+                    version = 6; // It is probably version 6
+                    if ((msg[2].length() != 0 && !msg[2].equals("HF-LPB100"))) {
```

Comment:
The argument can be the literal or an empty string. Depending on the firmware version.